### PR TITLE
feature(file-input-components): use components from `@angular/material` instead of our own deprecated ones

### PR DESCRIPTION
**Pull request by niwolf**
@plentymarkets/team-terra

#### Must be done by Terra

Testing

-   [ ] Person 1 (mandatory)

    Browser support (relevant for changes in scss / appearance of components)

    -   [ ] Chrome
    -   [ ] Firefox
    -   [ ] Safari

    Responsiveness

    -   [ ] Desktop
    -   [ ] Tablet
    -   [ ] Mobile

-   [ ] Person 2 (mandatory)

    Browser support

    -   [ ] Chrome
    -   [ ] Firefox
    -   [ ] Safari

    Responsiveness

    -   [ ] Desktop
    -   [ ] Tablet
    -   [ ] Mobile

Information transfer

-   [ ] Inform Team plentyShop about changes in `TerraFormComponent` (optional: relevant for changes in this component)

Terra Basic Plugin

-   [ ] Adapt changes (optional: relevant for version updates, global design changes, l10n)

---

#### Must be done by every developer

Please inform a member of Terra (@plentymarkets/team-terra) to get the upper part of the checklist done (with urgency or deadline).

Documentation

-   [ ] Changelog (optional: relevant for every change which influences the API)
-   [ ] Example (updated or created)
-   [ ] JSDoc (optional: relevant for every change which influences the API)
-   [ ] [Google spreadsheet](https://docs.google.com/spreadsheets/d/1OINnux8TEoitV-qdAxqUQaf9oGI_fqFVwfRWenRFfhI/edit#gid=0) (relevant when deprecating public APIs or features)

Testing

-   [ ] Unit Test (updated or created)

### CODE REVIEW

File: src/lib/components/forms/input/file-input/terra-file-input.component.scss [hunk 1]
```diff
@@ -3,40 +3,5 @@
...
-            > [class^='icon-'] {
-                display: inline-flex;
-                margin: 0 auto;
-                font-size: 7.5em;
-
-                > [class^='path']:before {
-                    color: var(--color-structure-9);
-                }
-            }
-        }
```
Comment (pweyrich) [inline src/lib/components/forms/input/file-input/terra-file-input.component.scss:36]: Can these lines really be removed?
The file-preview is still exisiting

File: src/lib/components/forms/input/file-input/terra-file-input.component.scss [hunk 2]
```diff
@@ -3,40 +3,5 @@
 .file-select {
     .file-preview {
         flex: 1;
-        width: 100%;
```
Comment (pweyrich) [inline src/lib/components/forms/input/file-input/terra-file-input.component.scss:6]: The import in line 1 should be removed!

File: src/lib/components/forms/input/file-input/terra-file-input.component.scss (hunk 1 above)
Comment (niwolf) [reply src/lib/components/forms/input/file-input/terra-file-input.component.scss:36]: @adumitrescu-plenty

Comment (niwolf) [reply src/lib/components/forms/input/file-input/terra-file-input.component.scss:36]: This was not included in the first merge. 
Bildschirmfoto 2021-11-02 um 15 00 31

I'll replace it back

Comment (adumitrescu-plenty) [reply src/lib/components/forms/input/file-input/terra-file-input.component.scss]: Oh right, I deleted more than I should have. Thanks

File: src/lib/components/file-browser/image-preview/image-preview.component.ts [hunk 3]
```diff
@@ -26,7 +25,7 @@ export class TerraImagePreviewComponent {
         this._metadata = {};
         this._isLoading = true;
         if (
-            !isNullOrUndefined(object) &&
+            !(object === null || object === undefined) &&
```
Comment (OliverBhm) [inline src/lib/components/file-browser/image-preview/image-preview.component.ts:28]:
```suggestion
            object != null &&
```

File: src/lib/components/file-browser/terra-file-browser.component.ts [hunk 4]
```diff
@@ -32,13 +32,26 @@ export class TerraFileBrowserComponent extends TerraFileBrowser implements OnCha
...
     > = new EventEmitter();
 
     public _rightColumnWidth: number = 0;
     public _centerColumnWidth: number = 10;
+    public _currentSelectedNode: TerraFileBrowserNode;
+
+    public _hasChild: (_: number, node: TerraFileBrowserNode) => boolean = (
+        _: number,
+        node: TerraFileBrowserNode
+    ): boolean => !!node.children && node.children.length > 0;
```
Comment (OliverBhm) [inline src/lib/components/file-browser/terra-file-browser.component.ts:54]:
```suggestion
    ): boolean => node?.children?.length > 0;
```

File: src/lib/components/file-browser/image-preview/image-preview.component.ts (hunk 3 above)
Comment (nappynapster) [reply src/lib/components/file-browser/image-preview/image-preview.component.ts:28]: or simply `if(object)`

**Review by Neill83 — DISMISSED** [review]

**Review by nappynapster — DISMISSED** [review]

**Review by Neill83 — APPROVED** [review]

**Review by pweyrich — APPROVED** [review]
🥳 Great job guys!

**Review by JanikPlenty — APPROVED** [review]

**Pull request merged**